sklearn/datasets/_samples_generator.py

```diff
@@ -621,8 +624,16 @@ def make_circles(n_samples=100, shuffle=True, noise=None, random_state=None,
     if factor >= 1 or factor < 0:
         raise ValueError("'factor' has to be between 0 and 1.")
 
-    n_samples_out = n_samples // 2
-    n_samples_in = n_samples - n_samples_out
+    if isinstance(n_samples, (int, np.integer)):
+        n_samples_out = n_samples // 2
+        n_samples_in = n_samples - n_samples_out
+    else:
+        try:
+            n_samples_out, n_samples_in = n_samples
+        except ValueError:
+            raise ValueError('`n_samples` can be either an int or '
+                             'a two-element tuple. You passed ' +
+                             str(n_samples) + ' instead.')
```

Comment:
Is it possible that users pass strange things and str(n_samples) fail? I guess we don't need to output n_samples here.

Reply:
I would have to be really bizzare so that `str(...)` fails. Anyway, removed as requested.

Reply:
I was just inspired by brilliant keras error messages, for example [this one](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/engine/sequential.py#L159)